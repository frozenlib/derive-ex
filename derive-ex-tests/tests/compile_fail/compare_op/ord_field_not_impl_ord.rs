use derive_ex::derive_ex;

#[derive(PartialOrd, Eq, PartialEq)]
struct NotOrd;

#[derive_ex(Ord, PartialOrd, Eq, PartialEq)]
struct X {
    x: NotOrd,
}

fn main() {}
