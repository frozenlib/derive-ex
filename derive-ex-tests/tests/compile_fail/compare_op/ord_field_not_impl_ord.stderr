error[E0277]: the trait bound `NotOrd: Ord` is not satisfied
 --> tests/compile_fail/compare_op/ord_field_not_impl_ord.rs:8:5
  |
8 |     x: NotOrd,
  |     ^ the trait `Ord` is not implemented for `NotOrd`
  |
help: consider annotating `NotOrd` with `#[derive(Ord)]`
  |
4 + #[derive(Ord)]
5 | struct NotOrd;
  |
